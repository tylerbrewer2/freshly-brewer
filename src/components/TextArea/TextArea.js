import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// Style Helpers
import Fonts from '../../styles/helpers/Fonts';

const TextAreaCore = ({ className, identifier, onChange, ...rest }) => {
  return (
    <textarea
      className={className}
      name={identifier}
      id={identifier}
      onChange={onChange}
      {...rest}
    />
  )
}

TextAreaCore.propTypes = {
  className: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  identifier: PropTypes.string,
}

TextAreaCore.defaultProps = {
  identifier: null,
}

const TextArea = styled(TextAreaCore)`
  font-family: ${Fonts.body};
  font-size: 18px;
  width: 100%;
  border: 1px solid #4A4A4A;
  border-radius: 3px;
  padding: 10px;
  height: 100px;
`;

export default TextArea;
