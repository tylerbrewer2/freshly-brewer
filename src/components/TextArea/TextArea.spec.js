import React from 'react';
import renderer from 'react-test-renderer';

import TextArea from './TextArea';

describe('component structure', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <TextArea
        identifier="identifier"
        value="test value"
        onChange={jest.fn()}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
