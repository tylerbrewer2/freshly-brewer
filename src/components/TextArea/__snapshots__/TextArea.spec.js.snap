exports[`component structure renders correctly 1`] = `
<textarea
  className="sc-bdVaJa cOxvEZ"
  id="identifier"
  name="identifier"
  onChange={[Function]}
  value="test value" />
`;
